In [60]:
# Загружаем библтотеки

import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)


In [149]:
# Загрузка файла:
flights = pd.read_csv('Flights_Anapa.csv',skiprows = [0], header = None)
flights.columns = ["flight_id","city_departure","city_arrival","aircraft_model","flight_time","fuel_tax",
                   "flights_amount","profit","b_pass_psent","b_count","e_pass_psent","e_count","tot_pass_psent",
                   "holiday_mark","date_flight","week_part","weekend_or_workday"]
flights.head()

flight_id city_departure city_arrival       aircraft_model  flight_time  \
0     136642          Anapa     Belgorod  Sukhoi Superjet-100     0.833333   
1     136352          Anapa       Moscow       Boeing 737-300     1.666667   
2     136253          Anapa       Moscow       Boeing 737-300     1.666667   
3     136122          Anapa       Moscow       Boeing 737-300     1.666667   
4     136360          Anapa       Moscow       Boeing 737-300     1.666667   

   fuel_tax  flights_amount     profit  b_pass_psent  b_count  e_pass_psent  \
0   55392.0        531000.0   475608.0            83       10            63   
1  156400.0       1372400.0  1216000.0            83       10            69   
2  156400.0       1407800.0  1251400.0            91       11            69   
3  156400.0       1431000.0  1274600.0            83       10            73   
4  156400.0       1455400.0  1299000.0            91       11            72   

   e_count  tot_pass_psent  holiday_mark date_flight  week_part  \
0       54              65         False  2017-01-30        1.0   
1       82              70         False  2016-12-10        6.0   
2       82              71         False  2016-12-24        6.0   
3       87              73         False  2017-01-08        0.0   
4       86              73         False  2017-02-05        0.0   

  weekend_or_workday  
0     workday_flight  
1     weekend_flight  
2     weekend_flight  
3     weekend_flight  
4     weekend_flight

In [33]:
# Сделаем выборку малоприбыльных рейсов 
flights[['flight_id', 'profit']][flights.aircraft_model == 'Sukhoi Superjet-100'].sort_values('profit')

flight_id    profit
0       136642  475608.0
150     136807  475608.0
17      136844  519708.0
18      136887  539808.0
9       136922  552408.0
..         ...       ...
137     136900  709908.0
138     136841  709908.0
140     136857  709908.0
133     136755  709908.0
178     136801  709908.0

[90 rows x 2 columns]

In [81]:
# Посмотрим на распределение прибыли c рейсов
flights['profit'][(flights.aircraft_model == 'Boeing 737-300')].value_counts(bins=4)

(1344400.0, 1472800.0]      32
(1601200.0, 1729600.0]      23
(1472800.0, 1601200.0]      22
(1215486.399, 1344400.0]    13
Name: profit, dtype: int64

In [82]:
flights['profit'][(flights.aircraft_model == 'Sukhoi Superjet-100')].value_counts(bins=4)

(651333.0, 709908.0]      54
(592758.0, 651333.0]      20
(534183.0, 592758.0]      13
(475373.699, 534183.0]     3
Name: profit, dtype: int64

In [144]:
flights_Sj = flights[(flights.aircraft_model == 'Sukhoi Superjet-100')]
flights_Bg = flights[(flights.aircraft_model == 'Boeing 737-300')]

In [132]:
# Оценим 5 % рейсов с минимальной прибылью для самолетов Sukhoi Superjet-100
min_profit_Sj = int(np.percentile(flights_Sj['profit'], 5))
flights_Sj[flights_Sj['profit'] <= min_profit_Sj].sort_values('profit')

flight_id city_departure city_arrival       aircraft_model  flight_time  \
0       136642          Anapa     Belgorod  Sukhoi Superjet-100     0.833333   
150     136807          Anapa     Belgorod  Sukhoi Superjet-100     0.833333   
17      136844          Anapa     Belgorod  Sukhoi Superjet-100     0.833333   
18      136887          Anapa     Belgorod  Sukhoi Superjet-100     0.833333   
9       136922          Anapa     Belgorod  Sukhoi Superjet-100     0.833333   

     fuel_tax  flights_amount    profit  b_pass_psent  b_count  e_pass_psent  \
0     55392.0        531000.0  475608.0            83       10            63   
150   55392.0        531000.0  475608.0            66        8            70   
17    55392.0        575100.0  519708.0            50        6            85   
18    55392.0        595200.0  539808.0            66        8            82   
9     55392.0        607800.0  552408.0            83       10            77   

     e_count  tot_pass_psent  holiday_mark date_flight  week_part  \
0         54              65         False  2017-01-30        1.0   
150       60              69          True  2017-02-23        4.0   
17        73              80         False  2017-02-28        2.0   
18        70              80         False  2017-01-20        5.0   
9         66              77         False  2017-02-11        6.0   

    weekend_or_workday  
0       workday_flight  
150     workday_flight  
17      workday_flight  
18      workday_flight  
9       weekend_flight

Здесь мы видим, что у данных рейсов загруженность варьируется от 65% до 80%. Также мы видим, что есть рейсы с загрузкой 80% и с маленькой прибылью, в данных рейсах загруженность эконом класса составляет 82-85%, а бизнес каллса 55-66%, данный факт существенно влияет на прибыль. Выделяется рейс 136807 в праздничный день по среди недели с минимальной прибылью. 
День недели заметно  не влияет на прибыльность данных рейсов. 
Данные рейсы следует отменить, в виду малой рентабельности и загрузки.

In [147]:
# Оценим 10 % рейсов с минимальной прибылью для самолетов Boeing 737-300
min_profit_Bg = int(np.percentile(flights_Bg['profit'], 15))
flights_Bg[flights_Bg['profit'] <= min_profit_Bg].sort_values('profit')

flight_id city_departure city_arrival  aircraft_model  flight_time  \
1       136352          Anapa       Moscow  Boeing 737-300     1.666667   
8       136464          Anapa       Moscow  Boeing 737-300     1.666667   
2       136253          Anapa       Moscow  Boeing 737-300     1.666667   
5       136418          Anapa       Moscow  Boeing 737-300     1.666667   
6       136250          Anapa       Moscow  Boeing 737-300     1.666667   
3       136122          Anapa       Moscow  Boeing 737-300     1.666667   
7       136178          Anapa       Moscow  Boeing 737-300     1.666667   
10      136306          Anapa       Moscow  Boeing 737-300     1.666667   
152     136397          Anapa       Moscow  Boeing 737-300     1.666667   
4       136360          Anapa       Moscow  Boeing 737-300     1.666667   
154     136204          Anapa       Moscow  Boeing 737-300     1.666667   
24      136383          Anapa       Moscow  Boeing 737-300     1.666667   
151     136202          Anapa       Moscow  Boeing 737-300     1.666667   
11      136491          Anapa       Moscow  Boeing 737-300     1.666667   
25      136486          Anapa       Moscow  Boeing 737-300     1.666667   

     fuel_tax  flights_amount     profit  b_pass_psent  b_count  e_pass_psent  \
1    156400.0       1372400.0  1216000.0            83       10            69   
8    156400.0       1398000.0  1241600.0            58        7            78   
2    156400.0       1407800.0  1251400.0            91       11            69   
5    156400.0       1407800.0  1251400.0            75        9            74   
6    156400.0       1407800.0  1251400.0            66        8            77   
3    156400.0       1431000.0  1274600.0            83       10            73   
7    156400.0       1434600.0  1278200.0            75        9            76   
10   156400.0       1443200.0  1286800.0            66        8            79   
152  156400.0       1444400.0  1288000.0            58        7            82   
4    156400.0       1455400.0  1299000.0            91       11            72   
154  156400.0       1457800.0  1301400.0            50        6            85   
24   156400.0       1493200.0  1336800.0            66        8            83   
151  156400.0       1495600.0  1339200.0            91       11            75   
11   156400.0       1505400.0  1349000.0            83       10            78   
25   156400.0       1505400.0  1349000.0            66        8            83   

     e_count  tot_pass_psent  holiday_mark date_flight  week_part  \
1         82              70         False  2016-12-10        6.0   
8         93              76         False  2017-01-09        1.0   
2         82              71         False  2016-12-24        6.0   
5         88              74         False  2016-12-07        3.0   
6         91              75         False  2017-02-08        3.0   
3         87              73         False  2017-01-08        0.0   
7         90              75         False  2017-01-29        0.0   
10        94              77         False  2016-12-04        0.0   
152       97              79          True  2016-12-25        0.0   
4         86              73         False  2017-02-05        0.0   
154      101              81          True  2017-02-23        4.0   
24        98              81         False  2017-01-25        3.0   
151       89              76          True  2017-01-05        4.0   
11        93              78         False  2016-12-08        4.0   
25        99              81         False  2017-02-02        4.0   

    weekend_or_workday  
1       weekend_flight  
8       workday_flight  
2       weekend_flight  
5       workday_flight  
6       workday_flight  
3       weekend_flight  
7       weekend_flight  
10      weekend_flight  
152     weekend_flight  
4       weekend_flight  
154     workday_flight  
24      workday_flight  
151     workday_flight  
11      workday_flight  
25      workday_flight

Здесь мы видим, что у данных рейсов загруженность варьируется от 70% до 79%. В данных видно, что рейсы в определенные выходные дни малоприбыльны. Также тут представлены рейсы в выходные дни за неделю до Нового года,  также предрабочий выходной рейс 8 января. По моему мнению данные рейсы также следует отменить в виду маленькой прибыльности и минимальной загрузки рейсов.
